src/components/CreateContactSelector.tsx

```diff
@@ -0,0 +1,184 @@
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuPortal,
+  DropdownMenuTrigger,
+} from '@radix-ui/react-dropdown-menu';
+import { type SelectContact } from '@src/server/db/models';
+import { type Dispatch, useReducer } from 'react';
+import {
+  Discord,
+  Email,
+  Facebook,
+  Instagram,
+  Twitter,
+  Website,
+  Youtube,
+  type logoProps,
+} from './ContactIcons';
+import {
+  type Control,
+  type UseFormRegister,
+  useFieldArray,
+  type FieldErrors,
+} from 'react-hook-form';
+import { type z } from 'zod';
+import { type createClubSchema } from '@src/utils/formSchemas';
+
+type Contact = Omit<SelectContact, 'clubId'>;
+
+function Reducer(
+  state: Array<Contact['platform']>,
+  action: {
+    type: 'add' | 'remove';
+    target: Contact['platform'];
+  },
+) {
+  switch (action.type) {
+    case 'remove':
+      return state.filter((x) => x != action.target);
+    case 'add':
+      return [...state, action.target];
+  }
+}
+const startContacts: Array<Contact['platform']> = [
+  'discord',
+  'instagram',
+  'website',
+  'email',
+  'twitter',
+  'facebook',
+];
+
+type ContactSelectorProps = {
+  control: Control<z.infer<typeof createClubSchema>>;
+  register: UseFormRegister<z.infer<typeof createClubSchema>>;
+  errors: FieldErrors<z.infer<typeof createClubSchema>>;
+};
+const ContactSelector = ({
+  control,
+  register,
+  errors,
+}: ContactSelectorProps) => {
+  const { fields, append, remove } = useFieldArray({
+    control: control,
+    name: 'contacts',
+  });
+  const [available, dispatch] = useReducer(Reducer, startContacts);
+  const addNew = (platform: Contact['platform']) => {
+    dispatch({ type: 'remove', target: platform });
+    append({ platform: platform, url: '' });
+  };
+  const removeItem = (index: number, platform: Contact['platform']) => {
+    remove(index);
+    dispatch({ type: 'add', target: platform });
+  };
```

Comment:
Prob rename these functions

Reply:
I renamed them so hopefully that makes it more clear.